### PR TITLE
update troubleshooting for Flink 1.12

### PR DESCRIPTION
**Pull request by alpinegizmo**
I have

* removed TimeCharacteristic
* stopped using timeWindow
* rebased and included the commit from https://github.com/ververica/flink-training/pull/2 that reworks the troubled streaming job solutions to use WatermarkStrategy

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -36,12 +36,12 @@ subprojects {
...
     // artifact properties
     group = 'org.apache.flink'
-    version = '1.11-SNAPSHOT'
+    version = '1.12-SNAPSHOT'
     description = """Flink Training Exercises"""
 
     ext {
         javaVersion = '1.8'
-        flinkVersion = '1.11.2'
+        flinkVersion = '1.12.0'
```
Comment (NicoK) [inline build.gradle:44]: let's increase that to 1.12.2 while we're changing it
```suggestion
        flinkVersion = '1.12.2'
```

**Review by NicoK — CHANGES_REQUESTED** [review]
Thanks for these updates, I also ran the troubled streaming job in our troubleshooting test VM with VVP 2.4 and it shows roughly the same numbers in this quick test.

Can you add two more things?

1. update to 1.12.2
2. remove the remaining `env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)` in some of the scala exercises/solutions (hourly tips + long rides - so this should probably go into apache/flink-training as well)

**Comment by alpinegizmo** [comment]
The time characteristic was already removed from the scala exercises and solutions. Perhaps you are just noticing what's in the bin directories.

**Review by NicoK — APPROVED** [review]
Thanks @alpinegizmo, you were right - I still had some bin directories that `gradlew clean` didn't clean up

**Pull request merged**